LBHFSSPublicAPI/terraform/production/main.tf

```diff
@@ -14,49 +14,58 @@ provider "aws" {
-}
-
 terraform {
   backend "s3" {
     bucket  = "terraform-state-production-apis"
     encrypt = true
     region  = "eu-west-2"
-    key     = services/YOUR API NAME/state #e.g. "services/transactions-api/state"
+    key     = services/fss-public-api/state
   }
 }
 
-module "development" {
-  # Delete as appropriate:
-  source                      = "github.com/LBHackney-IT/aws-hackney-components-per-service-terraform.git//modules/environment/backend/fargate"
-  # source = "github.com/LBHackney-IT/aws-hackney-components-per-service-terraform.git//modules/environment/backend/ec2"
-  cluster_name                = "production-apis"
-  ecr_name                    = ecr repository name # Replace with your repository name - pattern: "hackney/YOUR APP NAME"
-  environment_name            = "production"
-  application_name            = local.application_name 
-  security_group_name         = back end security group name # Replace with your security group name, WITHOUT SPECIFYING environment. Usually the SG has the name of your API
-  vpc_name                    = "vpc-production-apis"
-  host_port                   = port # Replace with the port to use for your api / app
-  port                        = port # Replace with the port to use for your api / app
-  desired_number_of_ec2_nodes = number of nodes # Variable will only be used if EC2 is required. Do not remove it. 
-  lb_prefix                   = "nlb-production-apis"
-  ecs_execution_role          = data.aws_iam_role.ecs_task_execution_role.arn
-  lb_iam_role_arn             = data.aws_iam_role.ec2_container_service_role.arn
-  task_definition_environment_variables = {
-    ASPNETCORE_ENVIRONMENT = "production"
-  }
-  task_definition_environment_variable_count = number # This number needs to reflect the number of environment variables provided
-  cost_code = your project's cost code
-  task_definition_secrets      = {}
-  task_definition_secret_count = number # This number needs to reflect the number of environment variables provided
+/*    POSTGRES SET UP    */
+data "aws_vpc" "production_vpc" {  
+  tags = {    
+    Name = "vpc-production-apis-production"  
+    }
+}
+data "aws_subnet_ids" "production_private_subnets" {  
+  vpc_id = data.aws_vpc.production_vpc.id  
+  filter {    
+    name   = "tag:Type"    
+    values = ["private"]  
+    }
+}
+
+ data "aws_ssm_parameter" "fss_public_postgres_db_password" {
+   name = "/fss-public-api/production/postgres-password"
+ }
+
+ data "aws_ssm_parameter" "fss-public_postgres_username" {
+   name = "/fss-public-api/production/postgres-username"
+ }
+
+module "postgres_db_production" {
+  source = "github.com/LBHackney-IT/aws-hackney-common-terraform.git//modules/database/postgres"
+  environment_name = "production"
+  vpc_id = data.aws_vpc.production_vpc.id
+  db_engine = "postgres"
+  db_engine_version = "11.1"
+  db_identifier = "fss-public-dev-db"
+  db_instance_class = "db.t2.micro"
+  db_name = "fss-public_dev"
+  db_port  = 5302
```

Comment:
I see you are using a none standart port for the db here?  This port may need to be opened in the security group.  Also in the serverless config you are referencing an SSM parameter for the port.  Might it be a good idea to use that reference here as well to keep it consistent?  I imagine we will have to create these parameters for the deployment to work.

Reply:
Also, if the DB is going to be created in our AWS API accounts, please use a port number commencing with 6, rather than 5, to avoid conflicts with databases created as part of the Platform APIs project

Reply:
Thanks for the input both. Just to clarify my PR message, the values for terraform were taken from the file below after speaking with Selwyn:
https://github.com/LBHackney-IT/resident-contact-api/blob/master/terraform/development/main.tf

(other file changes were taken from LBHackney-IT/housing-resident-information-api solution)

We are happy to use whatever config settings work for you. If you let me know the typical values I will update this branch. I don’t want to you to change from the conventions you presently use 👍

Reply:
Yes the example I provided was taken from the platform API.  I believe Mirela was saying that that project uses ports in the 5000 range and suggested we use a port in a higher range if we are deploying to the same AWS account, which I believe we are.